### ISSUE TITLE
Saving Texture3D resource through script does not work in Godot 4

### ISSUE THREAD

**Issue by Kwu564** [godotengine/godot#72993]
### Godot version

4.0.beta.3

### System information

Vulkan, Windows 10 GTX970m

### Issue description

Saving Texture3Ds from script is does not work in Godot 4. Someone apparently already fixed it but there's no pr of it yet it seems: https://github.com/Game3DEE/godot/commit/68bb829d49afcd4f0bd6516abbed84f1a27df61f 

Edit: Turns out there's already a pr here: https://github.com/godotengine/godot/issues/66558 but it's not merged yet.
### Steps to reproduce

First declare an array of type Image to hold each layer of the 3D Texture. 

Initialize the Image array in a triple nested for loop where the outermost loop iterates through each each Image creating a new Image for the current layer, and the two inner loops iterate through each Image's x and y pixel coordinates setting the pixel at those coordinates. 

Then, create an ImageTexture3D by first declaring ImageTexture3D.new() and then initializing it with ImageTexture3D.create() setting the format to Image.FORMAT_RGBA_8, its dimensions and passing in the image layer array to it. 

Finally save the 3D texture using ResourceSaver.save(). Saving as tres outputs a 1x1x1 pink texture and saving as png or tga causes the saver to not write anything to the disk.

### Minimal reproduction project

Create a new empty project and follow the steps to to reproduce.

**Comment by Zireael07** [comment]
Beta *3*, really? There's an RC already out, double check if the issue still exists.

**Comment by Kwu564** [comment]
It's still not working on the latest snapshot beta 17

**Comment by AThousandShips** [comment]
Do you mean PR #71394?

**Comment by Kwu564** [comment]
Yes that's it.

**Comment by Calinou** [comment]
Duplicate of https://github.com/godotengine/godot/issues/66558.

**Comment by PGComai** [comment]
I'm trying to understand the process of creating the data array for the image.

> First declare an array of type Image to hold each layer of the 3D Texture.

>Initialize the Image array in a triple nested for loop where the outermost loop iterates through each each Image creating a new Image for the current layer, and the two inner loops iterate through each Image's x and y pixel coordinates setting the pixel at those coordinates.

Could someone help me with this part? Before reading this, my code looked like this:

```
var arr = []
	for x in range(-bounds,bounds):
		arr.append([])
		for y in range(-bounds,bounds):
			arr[x+bounds].append([])
			for z in range(-bounds,bounds):
				var loc = Vector3(x,y,z)
				var item = get_cell_item(loc)
				if item == 0:
					# cell is full
					arr[x+bounds][y+bounds].append(Color(1,1,1,1))
				elif item == -1:
					arr[x+bounds][y+bounds].append(Color(0,0,0,1))
```
How do I create an array of type image? And is this the correct loop arrangement?

**Comment by Kwu564** [comment]
The array should be used to store individual images. The outer most loop appends a new image to the image array. Then an inner loop iterates through each row of the current image and then the innermost loop iterates through each column of that current row. Once done you can create a new Texture3D passing along the array of images to it.

**Comment by PGComai** [comment]
> A Texture3D is just an array of 2d images. The array should be used to store individual images. The outer most loop appends a new image to the image array. Then an inner loop iterates through each row of the current image and then the innermost loop iterates through each column of that current row setting each pixel of that image. Once done you can create a new Texture3D passing along the array of images to it.

Thank you! This solved my problem, here is the new code:

```
func make_array_from_gm():
	var arr = []
	for z in range(-bounds,bounds):
		var arrimg = Image.create(2*bounds, 2*bounds, false, Image.FORMAT_L8)
		for x in range(-bounds,bounds):
			for y in range(-bounds,bounds):
				var loc = Vector3(x,y,z)
				var item = get_cell_item(loc)
				if item == 0:
					# cell is full
					arrimg.set_pixelv(Vector2i(x+bounds,y+bounds), Color.WHITE)
				elif item == -1:
					arrimg.set_pixelv(Vector2i(x+bounds,y+bounds), Color.BLACK)
		arr.append(arrimg)
	return arr
```

**Comment by DrLutzi** [comment]
I believe this issue deserves re-opening, as PR #71394 does not fix the problem for ImageTexture3D in 4.1.1 ([a278c1b](https://github.com/godotengine/godot/commit/a278c1b98a81738a35b96a933a6e6cf771f9ab2d))

The following code should produce a tres file containing the data. Instead, it produces an tres file without data:

```
var i := Image.create(1, 1, false, Image.FORMAT_RGB8)
i.set_pixel(0,0, Color(1,0,0))
var l := ImageTexture3D.new()
l.create(Image.FORMAT_RGB8, 1, 1, 1, false, [ i ])
ResourceSaver.save(l, "res://texture3d.tres")
```

The equivalent for Texture2DArray works fine:

```
var i := Image.create(1, 1, false, Image.FORMAT_RGB8)
i.set_pixel(0,0, Color(1,0,0))
var l := Texture2DArray.new()
l.create_from_images([ i ])
ResourceSaver.save(l, "res://layered.tres")
```

I'm not sure if this is really how I'm meant to save as I am still fairly unfamiliar with godot, but it does feel like there is an issue there.

**Comment by Kwu564** [comment]
Yeah, I agree. I'm thinking of contributing a fix if no one gets to it yet.

**Comment by q8f13** [comment]
Can confirm that ImageTexture3D instance generated from code can't be saved into tres file

> I believe this issue deserves re-opening, as PR #71394 does not fix the problem for ImageTexture3D in 4.1.1 ([a278c1b](https://github.com/godotengine/godot/commit/a278c1b98a81738a35b96a933a6e6cf771f9ab2d))
> 
> The following code should produce a tres file containing the data. Instead, it produces an tres file without data:
> 
> ```
> var i := Image.create(1, 1, false, Image.FORMAT_RGB8)
> i.set_pixel(0,0, Color(1,0,0))
> var l := ImageTexture3D.new()
> l.create(Image.FORMAT_RGB8, 1, 1, 1, false, [ i ])
> ResourceSaver.save(l, "res://texture3d.tres")
> ```
> 
> The equivalent for Texture2DArray works fine:
> 
> ```
> var i := Image.create(1, 1, false, Image.FORMAT_RGB8)
> i.set_pixel(0,0, Color(1,0,0))
> var l := Texture2DArray.new()
> l.create_from_images([ i ])
> ResourceSaver.save(l, "res://layered.tres")
> ```
> 
> I'm not sure if this is really how I'm meant to save as I am still fairly unfamiliar with godot, but it does feel like there is an issue there.

Can confirm that ImageTexture3D generated from code can't be saved into valid tres/res file.
File generated as tres just be like:
`[gd_resource type="ImageTexture3D" format=3 uid="uid://1brwgfcw8yvu"]

[resource]
`
Using godot 4.1.1 bd6af8e0e

**Comment by AThousandShips** [comment]
This is a duplicate, please see:
* https://github.com/godotengine/godot/issues/66558
